zrml/prediction-markets/src/benchmarks.rs

```diff
@@ -397,47 +376,59 @@ benchmarks! {
+        let (caller, market_id) =
+            setup_resolve_common_categorical::<T>(a, b, c_u16)?;
+        T::MarketCommons::mutate_market(&market_id, |market| {
+            let admin = account("admin", 0, 0);
+            market.dispute_mechanism = MarketDisputeMechanism::Authorized(admin);
+            Ok(())
+        })?;
 
         for i in 0..c.min(d) {
             let origin = caller.clone();
-            let disputes = crate::Disputes::<T>::get(market_id);
-            let market = T::MarketCommons::market(&Default::default()).unwrap();
-            T::SimpleDisputes::on_dispute(&disputes, &market_id, &market)?;
+            Pallet::<T>::dispute(RawOrigin::Signed(origin).into(), market_id, OutcomeReport::Categorical(i.saturated_into::<u16>()))?;
         }
     }: {
         let market = T::MarketCommons::market(&market_id)?;
-        let disputes = crate::Disputes::<T>::get(market_id);
-        T::SimpleDisputes::on_resolution(&disputes, &market_id, &market)?
+        Pallet::<T>::on_resolution(&market_id, &market)?;
+    } verify {
+        let market = T::MarketCommons::market(&market_id)?;
+        assert_eq!(market.status, MarketStatus::Resolved);
     }
 
     internal_resolve_scalar_reported {
         let total_accounts = 10u32;
         let asset_accounts = 10u32;
-        let (_, market_id) = setup_resolve_common_scalar_after_dispute::<T>(total_accounts, asset_accounts)?;
+    let (caller, market_id) = setup_resolve_common_scalar::<T>(total_accounts, asset_accounts)?;
     }: {
         let market = T::MarketCommons::market(&market_id)?;
-        let disputes = crate::Disputes::<T>::get(market_id);
-        T::SimpleDisputes::on_resolution(&disputes, &market_id, &market)?
+        Pallet::<T>::on_resolution(&market_id, &market)?;
+    } verify {
+        let market = T::MarketCommons::market(&market_id)?;
+        assert_eq!(market.status, MarketStatus::Resolved);
     }
 
     internal_resolve_scalar_disputed {
         let total_accounts = 10u32;
         let asset_accounts = 10u32;
         let d in 0..T::MaxDisputes::get();
 
-        let (caller, market_id) = setup_resolve_common_scalar_after_dispute::<T>(total_accounts, asset_accounts)?;
-
+        let (caller, market_id) = setup_resolve_common_scalar::<T>(total_accounts, asset_accounts)?;
+        T::MarketCommons::mutate_market(&market_id, |market| {
+            let admin = account("admin", 0, 0);
+            market.dispute_mechanism = MarketDisputeMechanism::Authorized(admin);
+            Ok(())
+        })?;
         for i in 0..d {
-            let disputes = crate::Disputes::<T>::get(market_id);
             let origin = caller.clone();
-            let market = T::MarketCommons::market(&Default::default()).unwrap();
-            T::SimpleDisputes::on_dispute(&disputes, &market_id, &market)?;
+            Pallet::<T>::dispute(RawOrigin::Signed(origin).into(), market_id, OutcomeReport::Scalar(i.into()))?;
```

Comment:
You may want to add an `assert` which ensures that `d < range.end` where `range` is the scalar range.

```suggestion
            Pallet::<T>::dispute(
                RawOrigin::Signed(origin).into(),
                market_id,
                OutcomeReport::Scalar(i.into()),
            )?;
```

Reply:
range is u128::MAX so hopefully not needed, and to get it from market_type need to unfold enum.

Reply:
Sure. It's a nitpick, but you're still _assuming_ that that's the case. Not important, though.